src/features/jserrors/instrument/index.js

```diff
@@ -71,67 +62,55 @@ export class Instrument extends InstrumentBase {
   /** Restoration and resource release tasks to be done if JS error loader is being aborted. Unwind changes to globals. */
   #abort () {
     this.removeOnAbort?.abort()
+    this.#seenErrors.clear()
     this.abortHandler = undefined // weakly allow this abort op to run only once
   }
 
+  #castError (error) {
+    if (error instanceof Error) {
+      return error
+    }
+
+    if (typeof error.message !== 'undefined') {
+      return new UncaughtError(error.message, error.filename, error.lineno, error.colno)
+    }
+
+    return new UncaughtError(error)
+  }
+
   /**
-   * FF and Android browsers do not provide error info to the 'error' event callback,
-   * so we must use window.onerror
-   * @param {string} message
-   * @param {string} filename
-   * @param {number} lineno
-   * @param {number} column
-   * @param {Error | *} errorObj
-   * @returns
+   * Attempts to convert a PromiseRejectionEvent object to an Error object
+   * @param {PromiseRejectionEvent} unhandledRejectionEvent The unhandled promise rejection event
+   * @returns {Error} An Error object with the message as the casted reason
    */
-  onerrorHandler (message, filename, lineno, column, errorObj) {
-    if (typeof this.origOnerror === 'function') this.origOnerror(...arguments)
-
-    try {
-      if (this.skipNext) this.skipNext -= 1
-      else handle('err', [errorObj || new UncaughtException(message, filename, lineno), now()], undefined, FEATURE_NAMES.jserrors, this.ee)
-    } catch (e) {
+  #castPromiseRejectionEvent (promiseRejectionEvent) {
+    let prefix = 'Unhandled Promise Rejection: '
+    if (promiseRejectionEvent?.reason instanceof Error) {
       try {
-        handle('ierr', [e, now(), true], undefined, FEATURE_NAMES.jserrors, this.ee)
-      } catch (err) {
-        // do nothing
+        promiseRejectionEvent.reason.message = prefix + promiseRejectionEvent.reason.message
+        return promiseRejectionEvent.reason
+      } catch (e) {
+        return promiseRejectionEvent.reason
       }
     }
-    return false // maintain default behavior of the error event of Window
-  }
-}
-
-/**
- *
- * @param {string} message
- * @param {string} filename
- * @param {number} lineno
- */
-function UncaughtException (message, filename, lineno) {
-  this.message = message || 'Uncaught error with no additional information'
-  this.sourceURL = filename
-  this.line = lineno
-}
-
-/**
- * Attempts to cast an unhandledPromiseRejection reason (reject(...)) to an Error object
- * @param {*} reason - The reason property from an unhandled promise rejection
- * @returns {Error} - An Error object with the message as the casted reason
- */
-function castReasonToError (reason) {
-  let prefix = 'Unhandled Promise Rejection: '
-  if (reason instanceof Error) {
+    if (typeof promiseRejectionEvent.reason === 'undefined') return new Error(prefix)
```

Comment:
Should keep in mind that when we create a new Error, the stack trace is going to point back to this code (us), so users will likely first come to us saying there's some error being thrown in our code. 

That said, if they call `reject()` in promise without an arg, should this be more descriptive than just the prefix string?

Reply:
This code was not changed, just moved. That said, I also questioned why we were creating new Error instances here for the same reason. I thought about changing this to use the `UncaughtError` class.